Sources/Workspace/Workspace.swift

```diff
@@ -2270,8 +2270,12 @@ extension Workspace {
             }
         }
 
+        // download max n files concurrently
+        let semaphore = DispatchSemaphore(value: 8)
```

Comment:
why 8?

Reply:
we have `Concurrency.maxOperations` which maps to active cores, maybe we should use that or at least use that as the max ie `min(8, Concurrency.maxOperations)` if 8 is the right number

Reply:
8 was just a random number that I came up with. `Concurrency.maxOperations` makes a lot more sense, thank you for the hint!